subprojects/gradle/runner/device-provider/src/main/kotlin/com/avito/android/runner/devices/internal/kubernetes/KubernetesReservationClient.kt

```diff
@@ -85,9 +92,12 @@ internal class KubernetesReservationClient(
             namespace = kubernetesApi.namespace,
             reservation = reservation
         )
-        val deploymentName = deployment.metadata.name
-        deploymentsChannel.send(deploymentName)
-        kubernetesApi.createDeployment(deployment)
+        val deploymentName = lock.withLock {
+            val deploymentName = deployment.metadata.name
+            deploymentsChannel.send(deploymentName)
```

Comment:
Is it ok that we send a deployment before creation?

Reply:
I don't know I'm scared to change that)))